### PR TITLE
CRIMAP-173 Use sub-navigation in dashboard

### PR DESCRIPTION
**Pull request by kennyevil**
## Description of change

Adds subnavigation to the applications page

## Link to relevant ticket

https://dsdmoj.atlassian.net/browse/CRIMAP-173

## Screenshots of changes (if applicable)

0 0 0 0_3000_completed_applications

## How to manually test the feature

Complete and submit an application and then you should see it under submitted in 'you applications'. When you click the link for the application it should take you to the show page for that application. In developer tools, there will be an option to mark the application as 'returned'

### CODE REVIEW

File: config/locales/en/dashboard.yml [hunk 1]
```diff
@@ -16,6 +16,18 @@ en:
         in_progress: In progress
         submitted: Submitted
         returned: Returned
+      sub_navigation:
+        link_1: In progress %{count}
+        link_2: Submitted %{count}
+        link_3: Returned %{count}
+    show:
```
Comment (zheileman) [inline config/locales/en/dashboard.yml:23]: I think this `show` action is no longer in this controller but instead in the `completed_applications` one so we can remove these locales. Perhaps were reintroduced in a rebase, sorry about that!

File: app/views/completed_applications/index.html.erb [hunk 2]
```diff
@@ -1,3 +1,95 @@
...
+
+    <h1 class="govuk-heading-xl"><%=t '.heading' %></h1>
+
+    <%= button_to crime_applications_path,
+                  class: 'govuk-button govuk-button--start',
+                  data: { module: 'govuk-button' } do; t('.start_button'); end %>
+  </div>
+</div>
+
+<nav class="moj-sub-navigation" aria-label="Sub navigation">
```
Comment (zheileman) [inline app/views/completed_applications/index.html.erb:15]: I think we may be able to extract the sub navigation into its own partial and just pass as `locals` the different counters?
I've added another comment to make the code simpler too, leveraging the locales, with no need for if-else.

File: app/controllers/crime_applications_controller.rb [hunk 3]
```diff
@@ -1,14 +1,18 @@
 class CrimeApplicationsController < DashboardController
   before_action :check_crime_application_presence,
-                :present_crime_application, except: [:index, :create]
+                :present_crime_application, only: [:show, :edit, :destroy, :confirm_destroy]
```
Comment (zheileman) [inline app/controllers/crime_applications_controller.rb:3]: We moved the `#show` action into the other controller.

File: app/controllers/completed_applications_controller.rb [hunk 4]
```diff
@@ -2,24 +2,34 @@ class CompletedApplicationsController < DashboardController
...
+                        .includes(:applicant)
+                        .merge(CrimeApplication.order(created_at: :desc))
+                    end
+
+    @in_progress_applications_count = CrimeApplication.in_progress
+                                                      .joins(:people)
+                                                      .includes(:applicant)
+                                                      .merge(Applicant.with_name)
+                                                      .count
+    @submitted_applications_count = CrimeApplication.submitted.count
```
Comment (zheileman) [inline app/controllers/completed_applications_controller.rb:29]: Tiny thing but might be a big win on the long run and simplify a bit the code, we don't need counter for submitted applications as per design (as this counter may run into the thousands anyways and is not very helpful to providers).

File: app/controllers/completed_applications_controller.rb [hunk 5]
```diff
@@ -2,24 +2,34 @@ class CompletedApplicationsController < DashboardController
...
-    #
-    # We don't have yet the `returned` status, and all this is really
-    # a bit pointless as this controller will ONLY talk to the
-    # document store through an API, but we don't have any of that yet.
+    @applications = if params[:q] == 'returned'
+                      CrimeApplication
+                        .returned
+                        .joins(:people)
+                        .includes(:applicant)
+                        .merge(CrimeApplication.order(created_at: :desc))
```
Comment (zheileman) [inline app/controllers/completed_applications_controller.rb:15]: I think for submitted or returned we want this sorted by `submitted_at: :desc`

Agree with not over optimising these queries right now, as it all will change with the document store 👍

**Review by zheileman — CHANGES_REQUESTED** [review]
Looking very nice! I added a few comments, nothing major but can have a big impact on simplifying the code 👍

**Review by arthurashman — APPROVED** [review]
Awesome! Looks and works as expected. 

Agree with Jesus' suggestions

File: config/locales/en/dashboard.yml [hunk 6]
```diff
@@ -55,3 +87,14 @@ en:
...
+    subnavigation:
+        in_progress:
+          zero: In progress
+          one: In progress (1)
+          other: In progress (%{count})
+        submitted: Submitted
+        returned:
+          zero: Returned
+          one: Returned (1)
+          other: Returned %{count}
```
Comment (zheileman) [inline config/locales/en/dashboard.yml:100]: Missing parenthesis in the counter 😛

File: config/locales/en/dashboard.yml [hunk 7]
```diff
@@ -42,6 +52,28 @@ en:
...
+      table_headings:
+        applicant_name: Name
+        submitted_at: Submitted
+        reference: LAA reference
+        status: Status
+        action: Action
+      status:
+        in_progress: In progress
+        submitted: Submitted
+      sub_navigation:
```
Comment (zheileman) [inline config/locales/en/dashboard.yml:67]: Do we need these locales if we are using the partial?

File: config/locales/en/dashboard.yml [hunk 8]
```diff
@@ -16,6 +16,16 @@ en:
         in_progress: In progress
         submitted: Submitted
         returned: Returned
+      sub_navigation:
```
Comment (zheileman) [inline config/locales/en/dashboard.yml:19]: Same with this partials.

**Review by zheileman — APPROVED** [review]
👍 Cool. Just a tiny thing with the locales, but I'm approving anyways.
As part of another PR we can clean up the status column and tags as they are no longer used with this new design.

File: config/locales/en/dashboard.yml (hunk 6 above)
Comment (kennyevil) [reply config/locales/en/dashboard.yml:100]: *swearwords*

**Pull request merged**